src/conversions/rust_decimal.rs

```diff
@@ -0,0 +1,91 @@
+        //       exposed via the C API so its a bit funky to make those
+        //       exceptions directly
+        Decimal::from_str_exact(obj.str()?.to_str()?)
+            .map_err(|e| PyTypeError::new_err(e.to_string()))
+    }
+}
+
+impl ToPyObject for Decimal {
+    fn to_object(&self, py: Python<'_>) -> PyObject {
+        // TODO: how do we handle the error gracefully?
+        // look up the decimal.Decimal
+        let dec_mod = py
+            .import(intern!(py, "decimal"))
+            .expect("failed to import decimal");
+        let dec_cls = dec_mod
+            .getattr(intern!(py, "Decimal"))
+            .expect("failed getattr Decimal from decimal");
+        // now call the constructor with the Rust Decimal string-ified
+        // to not be lossy
+        // TODO: is to_string() the best conversion?
+        let ret = dec_cls
+            .call1((self.to_string(),))
+            .expect("failed to call decimal.Decimal(value)");
+        ret.to_object(py)
+    }
+}
+
+impl IntoPy<PyObject> for Decimal {
+    fn into_py(self, py: Python<'_>) -> PyObject {
+        self.to_object(py)
+    }
+}
+
+#[cfg(test)]
+mod test_rust_decimal {
+    use super::*;
+    use crate::types::PyDict;
+    use indoc::formatdoc;
+    use rust_decimal::Decimal;
+
+    macro_rules! convert_constants {
+        ($name:ident, $rs:expr, $py:literal) => {
+            #[test]
+            fn $name() {
+                Python::with_gil(|py| {
+                    let rs_orig = $rs;
+                    let rs_dec = rs_orig.into_py(py);
+                    let locals = PyDict::new(py);
+                    locals.set_item("rs_dec", &rs_dec).unwrap();
+                    // Checks if Rust Decimal -> Python Decimal conversion is correct
+                    py.run(
+                        &formatdoc! {
+                        r#"
+                            import decimal
+                            py_dec = decimal.Decimal({})
+                            assert py_dec == rs_dec
+                        "#, $py},
+                        None,
+                        Some(locals),
+                    )
+                    .unwrap();
+                    // Checks if Python Decimal -> Rust Decimal conversion is correct
+                    let py_dec = locals.get_item("py_dec").unwrap();
+                    let py_result: Decimal = FromPyObject::extract(py_dec).unwrap();
+                    assert_eq!(rs_orig, py_result);
+                })
+            }
+        };
+    }
+
+    convert_constants!(convert_zero, Decimal::ZERO, "0");
+    convert_constants!(convert_one, Decimal::ONE, "1");
+    convert_constants!(convert_neg_one, Decimal::NEGATIVE_ONE, "-1");
+    convert_constants!(convert_two, Decimal::TWO, "2");
+    convert_constants!(convert_ten, Decimal::TEN, "10");
+    convert_constants!(convert_one_hundred, Decimal::ONE_HUNDRED, "100");
+    convert_constants!(convert_one_thousand, Decimal::ONE_THOUSAND, "1000");
```

Comment:
It sounds like we need tests for infinity and NaN?

Reply:
There's no representation of Infinity however. Not sure on the best approach there.

Reply:
I think `Option<Decimal>` isn't appropriate here, because that really means Python `Decimal` or Python `None`.

Probably best to just make those cases return errors for the moment.

If someone needs NaN & infinity they could build a wrapper around `rust_decimal` and check for those values before using the conversion defined here.

Reply:
Added tests for `NaN` and `Infinity`